typing/typetexp.ml

```diff
@@ -970,11 +969,15 @@ let report_error env ppf = function
        longident lid
        (fun ppf -> may (fprintf ppf "@\n%a" Includemod.report_error) details)
   | Illegal_reference_to_recursive_module ->
-      fprintf ppf "Illegal recursive module reference"
+     fprintf ppf "Illegal recursive module reference"
   | Access_functor_as_structure lid ->
       fprintf ppf "The module %a is a functor, not a structure" longident lid
   | Apply_structure_as_functor lid ->
-      fprintf ppf "The module %a is a structure, not a functor" longident lid
+     fprintf ppf "The module %a is a structure, not a functor" longident lid
+  | Use_generative_functor_as_applicative flid ->
+     fprintf ppf "@[%a is a generative functor,@ and@ so@ cannot@ be@ applied@ \
+                    in@ type@ expressions@]"
+       longident flid
```

Comment:
That error message sounds wrong. Shouldn't be it `_ is a generative functor but was applied to _. It should be applied to ()` ?